src/core/drive/form_submission.ts

```diff
@@ -99,7 +99,9 @@ export class FormSubmission {
   }
 
   requestSucceededWithResponse(request: FetchRequest, response: FetchResponse) {
-    if (this.requestMustRedirect(request) && !response.redirected) {
+    if (response.unprocessableEntity) {
+      this.delegate.formSubmissionFailedWithResponse(this, response)
+    } else if (this.requestMustRedirect(request) && !response.redirected) {
       const error = new Error("Form responses must redirect to another location")
```

Comment:
What is the technical reason why we need to have this error trigger when the response is not a redirect?

Reply:
https://github.com/hotwired/turbo/issues/22#issuecomment-750336078

Reply:
> What is the technical reason why we need to have this error triggered when the response is not a redirect?

https://github.com/hotwired/turbo/issues/22#issuecomment-750336078